tipg/factory.py

```diff
@@ -1197,17 +1221,299 @@ def links(self, request: Request) -> List[model.Link]:
+
+            if output_type == MediaType.html:
+                return self._create_html_response(
+                    request,
+                    data.json(exclude_none=True),
+                    template_name="tilematrixsets",
+                )
+
+            return data
+
+        @self.router.get(
+            r"/tileMatrixSets/{tileMatrixSetId}",
+            response_model=TileMatrixSet,
+            response_model_exclude_none=True,
+            summary="Retrieve the definition of the specified tiling scheme (tile matrix set).",
+            operation_id="getTileMatrixSet",
+            responses={
+                200: {
+                    "content": {
+                        MediaType.html.value: {},
+                        MediaType.json.value: {},
+                    },
+                },
+            },
+        )
+        async def tilematrixset(
+            request: Request,
+            tileMatrixSetId: Literal[tuple(self.supported_tms.list())] = Path(
+                ...,
+                description="Identifier for a supported TileMatrixSet.",
+            ),
+            output_type: Optional[MediaType] = Depends(OutputType),
+        ):
+            """
+            OGC Specification: http://docs.opengeospatial.org/per/19-069.html#_tilematrixset
+            """
+            data = self.supported_tms.get(tileMatrixSetId)
+
+            if output_type == MediaType.html:
+                # For visualization purpose we add the tms bbox
+                data = {**data.dict(exclude_none=True), "bbox": data.bbox}
+                return self._create_html_response(
+                    request,
+                    json.dumps(data),
+                    template_name="tilematrixset",
+                )
+
+            return data
+
+        @self.router.get(
+            "/collections/{collectionId}/tiles",
+            response_model=model.TileSetList,
+            response_class=ORJSONResponse,
+            response_model_exclude_none=True,
+            responses={200: {"content": {MediaType.json.value: {}}}},
+            summary="Retrieve a list of available vector tilesets for the specified collection.",
+            operation_id=".collection.vector.getTileSetsList",
+        )
+        async def collection_tileset_list(
+            request: Request,
+            collection=Depends(self.collection_dependency),
+            output_type: Optional[MediaType] = Depends(OutputType),
+        ):
+            """Retrieve a list of available vector tilesets for the specified collection."""
+            collection_bbox = None
+            if bounds := collection.bounds:
+                collection_bbox = {
+                    "lowerLeft": [bounds[0], bounds[1]],
+                    "upperRight": [bounds[2], bounds[3]],
+                    "crs": "http://www.opengis.net/def/crs/OGC/1.3/CRS84",
+                }
+
+            data = model.TileSetList(
+                tilesets=[
+                    model.TileSet(
```

Comment:
in the `tileset_list` I choose not to add the `tileMatrixSetLimits` because it will make things way more complicated